### ISSUE TITLE
Missing trailing semicolon after at-rule causes incorrect warning for scss syntax

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#1043]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where warning for `block-closing-brace-newline-before` is returned rather than what should be a warning a missing semicolon with the `declaration-block-trailing-semicolon` rule.

> Which rule, if any, is this issue related to?

`declaration-block-trailing-semicolon` and `block-closing-brace-newline-before`

> What CSS is needed to reproduce this issue?

```css
a {
	@include font-size(14)
}
```
> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "block-closing-brace-newline-before": "always",
    "declaration-block-trailing-semicolon": "always"
  }
}
```

> Which version of stylelint are you using?

`5.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, using SCSS via `postcss-scss` syntax

> What did you expect to happen?

Expected to see the semi-colon warning:
 `Line 2, Col 24: Expected a trailing semicolon (declaration-block-trailing-semicolon)`

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:
`Line 2, Col 24: Expected newline before "}" (block-closing-brace-newline-before)`

If the CSS in fact does include the semicolon then no warnings are shown and as such works as expected, e.g:
```css
a {
	@include font-size(14);
}
```

**Comment by davidtheclark** [comment]
Weird one. I'd bet it has to do with the AST that results from this pattern (as opposed to the AST that results from the standard pattern of declarations inside rules).
